### DISCUSSION TITLE
Read document in document

### DISCUSSION THREAD

**Discussion by retwas** [SchneiderInfosystems/FB4D#115] [Q&A] [answered]
Hello,

I have a collection "users" with a lot of documents (user). 
For each user, I have collection "cards" with 0-n documents.

When I ask for user, I get the right document without any problem.
Then I can read user info (street, zipcode, birthdate, ..), but how can I get all the documents of the collection cards in my user ? 

Do I need to do a new call ?

Thanks.

**Comment by SchneiderInfosystems** [comment, marked as answer]
Hi, 

Yes, you need another get-call in order to fetch your subdocuments.

`IFirestoreDatabase.Get(['users', UserID, 'cards'], nil, OnReceiveCardDocs,  OnReceiveCarDocsError);
`
Alternatively, you can also use a listener you be informed when a document is changing. (`IFirestoreDatabase,SubscribeQuery`).

Thank you for giving FB4D a star!

**Reply by retwas** [reply]
Thanks you, perfect ! :)

**Comment by arvanus** [comment]
Hello, related to this question, what/how to do when you don't know the collection name (in this case `cards`)?
I have a case where I need to replicate recursively a collection, with N documents with N collections inside then (with N documents...) 
I can get the documents inside the collections I know the name, but can't figure out a way to get the coll. names
Thanks!

**Reply by SchneiderInfosystems** [reply]
This function is not yet available. As you can see, also most other SDKs from Google are not able to do this:
https://firebase.google.com/docs/firestore/query-data/get-data#list_subcollections_of_a_document
Usually, the path to a sub-document has a defined name (like cards). You can check if a sub-document exists but not list all sub-documents yet.

**Reply by arvanus** [reply]
Too bad, I was trying to bump/dump a database, but looks like I won't be able to do that now 
Thanks anyway for this wonderful library!
